### PR TITLE
Fix encoding of large numbers

### PR DESCRIPTION
**Pull request by jhermsmeier**
So, this is a fun one I noticed yesterday: Number.toString() can not be relied upon to give you all digits of an integer:

```js
// Number.toString()
> 340282366920938463463374607431768211456..toString()
'3.402823669209385e+38'

// Number.toPrecision() doesn't really help either
> 340282366920938463463374607431768211456..toPrecision(100)
'340282366920938463463374607431768211456.0000000000000000000000000000000000000000000000000000000000000'

// Neither does Number.toFixed()
> 340282366920938463463374607431768211456..toFixed(0)
'3.402823669209385e+38'

// Oh, look
> BigInt(340282366920938463463374607431768211456).toString()
'340282366920938463463374607431768211456'
```

While this is reasonably unlikely to happen in real-world usage as Number.toString() starts returning exponential notation at 2<sup>70</sup>, it still could occur, and considering this might be run on other engines than V8 which might have different limits, this could potentially occur sooner.

Unfortunately the only fix I found for this is going through `BigInt()` – I guess we could hand-roll something, but `BigInt()` seems to be the reliable thing to go for here – what do you think?

### CODE REVIEW

File: lib/encode.js [hunk 1]
```diff
@@ -63,21 +63,7 @@ encode.string = function (buffers, data) {
...
-
-  if (val !== data && !encode._floatConversionDetected) {
-    encode._floatConversionDetected = true
-    console.warn(
-      'WARNING: Possible data corruption detected with value "' + data + '":',
-      'Bencoding only defines support for integers, value was converted to "' + val + '"'
-    )
-    console.trace()
-  }
+  buffers.push(Buffer.from('i' + BigInt(data) + 'e'))
```
Comment (jhermsmeier) [inline lib/encode.js:66]: This is of course a breaking change, as `BigInt()` will through if you hand it a float or anything else invalid, which also made the float conversion warning obsolete

Comment (themasch) [reply lib/encode.js:66]: I think I am fine with a breaking change, especially if we reach support for the full v2 spec for that. 
And I did not know that BitInt is part of the core now, that neat!

**Review by themasch — APPROVED** [review]
Guess we'll have a bencode v3.0 then, soon?

**Comment by themasch** [comment]
I think this needs a new merge from master (with conflict solving in encode.js) and we need to remove the float test since this breaks with bigint.

**Comment by ThaUnknown** [comment]
superseeded by https://github.com/webtorrent/node-bencode/pull/150

**Pull request closed without merging**